### ISSUE TITLE
Query in using angle based surface normal Loss

### ISSUE THREAD

**Issue by rohun-tripathi** [xanderchf/MonoDepth-FPN-PyTorch#8]
Thanks you for your great work! 

When you calculate the normal loss, you do not take the absolute of the product per pixel before using torch.sum(). 
I think you should use the absolute before taking torch.sum().

For example, say an image only has two pixels. If the product of the predicted normal at the first pixel and the ground truth at the first pixel is -0.33 and the product of the predicted normal at the second pixel and the associated ground truth at the second pixel is 0.33, then your loss will be 0. Whereas, the loss is actually 0.33.

The code I am referring to is - 

> prod = ( grad_fake[:,:,None,:] @ grad_real[:,:,:,None] ).squeeze(-1).squeeze(-1)
> fake_norm = torch.sqrt( torch.sum( grad_fake**2, dim=-1 ) )
> real_norm = torch.sqrt( torch.sum( grad_real**2, dim=-1 ) )
